filebeat/scripts/generator/dashboards.go

```diff
@@ -0,0 +1,89 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package generator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/paths"
+)
+
+const (
+	dashboardEntry = `- id: %s
+  file: %s
+`
+)
+
+func AddModuleDashboard(beatName, module, kibanaVersion, dashboardID string, dashboard common.MapStr, suffix string) error {
+	version, err := common.NewVersion(kibanaVersion)
+	if err != nil {
+		return err
+	}
+	if version.Major < 6 {
+		return fmt.Errorf("saving exported dashboards is not available for Kibana version '%s'", version.String())
+	}
+
+	modulePath := filepath.Join(paths.Resolve(paths.Home, "module"), module)
+	stat, err := os.Stat(modulePath)
+	if err != nil || !stat.IsDir() {
+		return fmt.Errorf("no such module: %s\n", modulePath)
+	}
```

Comment:
If you want to report a more accurate error, I think you actually want to use https://golang.org/pkg/os/#IsExist instead of `IsDir()` or https://golang.org/pkg/os/#IsNotExist

Reply:
Why? If there is no error it means the file exists and I have the proper permission to access it. I also need to check if that module is a folder. If it is not a folder, I cannot create more folders under it and cannot saves files.

Reply:
I followed the same check as it is done in case of the `ModuleRegistry`: https://github.com/elastic/beats/blob/master/filebeat/fileset/modules.go#L116

Reply:
fair enough